File: thuat_toan_sinh.cpp
#include<iostream>

using namespace std;

int check(string s){
    for(int i=0;i<s.length()/2;i++)
        if(s[i]!=s[s.length()-1-i])
            return 0;
    return 1;
}

int main(){
    int n;
    cin>>n;
    string beg, last;
    for(int i=0;i<n;i++){
        beg+='0';
        last+='1';
        cout<<"0 ";
    }
    cout<<endl;
    while(beg!=last){
        for(int i=n-1;i>=0;i--){
            if(beg[i]=='1'){
                beg[i]='0';
                continue;
            }
            beg[i]='1';
            break;
        }
        if(check(beg)){
            for(int i=0;i<n;i++)
                cout<<beg[i]<<' ';
            cout<<endl;
        }
    }
    return 0;
}

File: thuat_toan_sinh.py
def check(s):
    for i in range(0,len(s)//2,1):
        if s[i] != s[len(s)-1-i]:
            return 0
    return 1

n = int(input())
beg, last = '', ''
for i in range(n):
    beg+='0'
    last+='1'
    print('0', end=' ')
print()
while beg!=last:
    for i in range(n-1,-1,-1):
        if beg[i]=='1':
            li=list(beg)
            li[i]='0'
            beg=''.join(li)
            continue
        li=list(beg)
        li[i]='1'
        beg=''.join(li)
        break
    if check(beg):
        for i in range(n):
            print(beg[i], end=' ')
        print()